Tests/SourceKitLSPTests/SemanticTokensTests.swift

```diff
@@ -0,0 +1,252 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import LanguageServerProtocol
+import LSPTestSupport
+import SKTestSupport
+import SourceKitLSP
+import XCTest
+
+final class SemanticTokensTests: XCTestCase {
+  /// Connection and lifetime management for the service.
+  var connection: TestSourceKitServer! = nil
+
+  /// The primary interface to make requests to the SourceKitServer.
+  var sk: TestClient! = nil
+
+  override func tearDown() {
+    sk = nil
+    connection = nil
+  }
+
+  override func setUp() {
+    connection = TestSourceKitServer()
+    sk = connection.client
+    _ = try! sk.sendSync(InitializeRequest(
+      processId: nil,
+      rootPath: nil,
+      rootURI: nil,
+      initializationOptions: nil,
+      capabilities: ClientCapabilities(
+        workspace: .init(
+          semanticTokens: .init(
+            refreshSupport: true
+          )
+        ),
+        textDocument: nil
+      ),
+      trace: .off,
+      workspaceFolders: nil
+    ))
+  }
+
+  private func performSemanticTokensRequest(text: String, range: Range<Position>? = nil) -> [SemanticToken] {
+    let url = URL(fileURLWithPath: "/\(#function)/a.swift")
+
+    // We wait for the first refresh request to make sure that the semantic tokens are ready
+
+    let expectation = XCTestExpectation(description: "performSemanticTokensRequest - refresh received")
```

Comment:
It’s better to use `self.expectation` to construct expectations and `self.wait` to wait for them. That way `XCTestCase` verifies that you didn’t set up any expectations but forgot to wait for them. `self.wait` will also fail the test case if the expectation wasn’t fulfilled in the timeout.